lib/writer/block-write-stream.js

```diff
@@ -134,10 +163,26 @@ class BlockWriteStream extends stream.Writable {
     }
 
     fs.write(this.fd, chunk, 0, chunk.length, chunk.position, (error, bytesWritten) => {
-      this.bytesWritten += bytesWritten
-      this.delta += bytesWritten
-      this.blocksWritten += 1
-      this.position += bytesWritten
+      if (!error) {
+        this.bytesWritten += bytesWritten
+        this.delta += bytesWritten
+        this.blocksWritten += 1
+        this.position += bytesWritten
+        this.retries = 0
+        next(error)
+        return
+      }
+
+      const isTransient = BlockWriteStream.isTransientError(error)
+
+      if (isTransient && (this.retries < this.maxRetries)) {
+        this.retries += 1
+        this._write(chunk, encoding, next)
+        return
+      } else if (isTransient) {
+        error.code = 'EUNPLUGGED'
```

Comment:
Should we make `EUNPLUGGED` a constant?

Reply:
I don't see much sense in that, tbh – and we don't do that for other error codes either.